tensorboard/webapp/runs/store/runs_reducers.ts

```diff
@@ -336,7 +336,33 @@ const {initialState: uiInitialState, reducers: uiNamespaceContextedReducers} =
         order: SortingOrder.DESCENDING,
       },
     },
-    {}
+    {},
+    /* onNavigated() */
+    (state, oldRoute, newRoute) => {
+      if (
+        !areSameRouteKindAndExperiments(oldRoute, newRoute) &&
+        newRoute.routeKind === RouteKind.COMPARE_EXPERIMENT &&
+        !state.runsTableHeaders.find(
+          (header) => header.name === 'experimentName'
+        )
+      ) {
+        const newRunsTableHeaders = [
+          ...state.runsTableHeaders,
+          {
+            type: ColumnHeaderType.EXPERIMENT,
+            name: 'experimentName',
+            displayName: 'Experiment',
+            enabled: true,
+          },
```

Comment:
This works for now. It should be noted this will always put the experiment column at the end of the list of columns. Eventually there will be other columns and I would think this should default right next to run. However, I am ok with dealing with that when we deal with all the ordering later.

I am going to be real curious how this will all work when adding experiments and hoping back and forth. Do the HParams reset every time? I am sure tomorrow we will get a thorough test of all that.